src/Internal/ResponseStream.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Http\Adapter\Artax\Internal;
+
+use Amp\Artax;
+use Amp\ByteStream\InputStream;
+use Amp\ByteStream\IteratorStream;
+use Amp\CancellationTokenSource;
+use Amp\CancelledException;
+use Amp\Emitter;
+use Amp\Promise;
+use Psr\Http\Message\StreamInterface;
+
+/** @internal */
```

Comment:
i would love a short comment what this is. something like "Stream wrapper to expose the amp stream as PSR-7 compatible stream."